src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/DoNotUseOutAttributeStringPInvokeParametersTests.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.DoNotUseOutAttributeStringPinvokeParametersAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.DoNotUseOutAttributeStringPinvokeParametersAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices.UnitTests
+{
+    public class DoNotUseOutAttributeStringPInvokeParametersAnalyzerTests
+    {
+        [Fact]
+        public async Task StringByReference_NoDiagnostics_Diagnostics_CS()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+
+public class C
+{
+    [DllImport(""user32.dll"", CharSet=CharSet.Unicode)]
+    private static extern void Method1(out string s); // OK
+
+    [DllImport(""user32.dll"", CharSet=CharSet.Unicode)]
+    private static extern void Method2([In] [Out] ref string s); // OK
+
+    [DllImport(""user32.dll"", CharSet=CharSet.Unicode)]
```

Comment:
Can you please add test to verify no diagnostic for method without DllImportAttribute?